edx_proctoring/views.py

```diff
@@ -464,22 +468,18 @@ def get(self, request):
         course_key = CourseKey.from_string(course_id)
         user = get_user_model().objects.get(username=(username or request.user.username))
         details = learning_sequences_service.get_user_course_outline_details(
-            course_key, user, pytz.utc.localize(datetime.max)
+            course_key, user, pytz.utc.localize(datetime.now())
```

Comment:
originally I was thinking this date pruned the results of `details.schedule`. Probably wrong on that. But for my sanity can you clarify how the date impacts that `details.schedule` object?

Reply:
Sure thing.

The `get_user_course_outline_details` method calls the [_get_user_course_outline_and_processors](https://github.com/edx/edx-platform/blob/351228cee40b5b0d51111d9a8c90a5432911838b/openedx/core/djangoapps/content/learning_sequences/api/outlines.py#L301) method. This runs the various processors on the course outline. The [ScheduleOutlineProcessor](https://github.com/edx/edx-platform/blob/351228cee40b5b0d51111d9a8c90a5432911838b/openedx/core/djangoapps/content/learning_sequences/api/processors/schedule.py#L19), which processes the start dates and due dates for exams, only extends the [inaccessible_sequences](https://github.com/edx/edx-platform/blob/351228cee40b5b0d51111d9a8c90a5432911838b/openedx/core/djangoapps/content/learning_sequences/api/processors/schedule.py#L58) method. It also includes a [schedule_data](https://github.com/edx/edx-platform/blob/351228cee40b5b0d51111d9a8c90a5432911838b/openedx/core/djangoapps/content/learning_sequences/api/processors/schedule.py#L109) method. It doesn't extend the `usage_keys_to_remove` method, which defines which usage keys to remove from the course outline. Note that [trimmed_course_outline](https://github.com/edx/edx-platform/blob/351228cee40b5b0d51111d9a8c90a5432911838b/openedx/core/djangoapps/content/learning_sequences/api/outlines.py#L354) in the [_get_user_course_outline_and_processors](https://github.com/edx/edx-platform/blob/351228cee40b5b0d51111d9a8c90a5432911838b/openedx/core/djangoapps/content/learning_sequences/api/outlines.py#L301) method is the result of removing the result of `usage_keys_to_remove` from the full course outline, and since the `ScheduleOutlineProcessor` doesn't extend this method, there are no keys to remove with respect to start dates and due dates. The `sections` attribute of the `UserCourseOutlineData` is the `sections` of the `trimmed_course_outline`, so I expect that the sections attribute will include exams that are yet to be released or that are past due.

The `schedule` attribute of `details` is determined by [calling](https://github.com/edx/edx-platform/blob/master/openedx/core/djangoapps/content/learning_sequences/api/outlines.py#L288) the `ScheduleOutlineProcessor`'s [schedule_data](https://github.com/edx/edx-platform/blob/351228cee40b5b0d51111d9a8c90a5432911838b/openedx/core/djangoapps/content/learning_sequences/api/processors/schedule.py#L109) method. The `schedule_data` method accepts as an argument the course outline returned by `_get_user_course_outline_and_processors`. It then iterates through the `sections` on `pruned_course_outline.sections` and creates the section classes from there. Because `sections` reflects the sections in the course outline with the `usage_keys_to_remove` removed, yet to be released and past due exams should still be in the `schedule`, since the `ScheduleOutlineProcessor` does not extend that method of the base class..

Does that help?